sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -274,16 +279,77 @@ protected override async ValueTask<Response> UpdateStatusAsync(bool async, Cance
             }
         }
 
-        private static Response GetResponseFromState(OperationState<T> state)
+        private static Response GetResponseFromState(OperationState<T> state, RequestMethod? requestmethod = null)
         {
             if (state.HasSucceeded)
             {
                 return state.RawResponse;
             }
 
+            // if this is a fake delete LRO with 404, just return empty response with 204
```

Comment:
Could we expand here or elsewhere on what a fake delete LRO is?

Reply:
added explanation for it

Reply:
This should apply for final get of delete LRO, moved this logic to `NextLinkOperationImplementation` instead, and added a test for it.